### ISSUE TITLE
issue a warning or throw exception when a combination of conditions is not supported by the query executor

### ISSUE THREAD

**Issue by luigidellaquila** [orientechnologies/orientdb#4306]
eg. in current release Lucene indexes in some cases cannot be safely used together with other conditions. Sometimes the system silently fails returning wrong results.
We should throw an exception in this case, at least until the correct behavior is implemented.

See issues #3683 #3640 #3611

**Comment by luigidellaquila** [comment]
added a validation layer just after query parsing, to verify that the WHERE condition matches minimum requirements to be calculated. 
Eg. parsing exception is thrown if 
- LUCENE filtering is used on a target that is not a class (eg. 
```sql 
select from (select out() from Foo where id = 4 ) where [prop1,prop2]  LUCENE "(prop1:foo AND props2:bar)"
```
- multiple LUCENE filters are defined in the same query (eg. 
```sql
select from Foo where prop1 LUCENE 'Foo' and prop2 LUCENE 'Bar'

Moving forward to next release because at this stage it's quite difficult to do a complete validation: it is executed in the core, while Lucene is a plugin, so the core misses some concepts. Some refactoring needed

**Comment by luigidellaquila** [comment]
@maggiolo00 shall I remove this check, now that in-memory Lucene indexes are implemented?

**Comment by maggiolo00** [comment]
@luigidellaquila yes we can remove that

**Comment by luigidellaquila** [comment]
Hi @maggiolo00,

it's already fixed, I'm closing it

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Errata, the fix will be released with 2.1.11
